RawSpeed/DngDecoderSlices.cpp

```diff
@@ -143,6 +143,159 @@ my_error_throw (j_common_ptr cinfo)
   ThrowRDE("JPEG decoder error!");
 } 
 
+static void decodeFPDeltaRow(unsigned char * src, unsigned char * dst, size_t tileWidth, size_t realTileWidth, unsigned int bytesps, int factor) {
+  // DecodeDeltaBytes
+  for (size_t col = factor; col < realTileWidth*bytesps; ++col) {
+    src[col] += src[col - factor];
+  }
+  // Reorder bytes into the image
+  // 16 and 32-bit versions depend on local architecture, 24-bit does not
+  if (bytesps == 3) {
+    for (size_t col = 0; col < tileWidth; ++col) {
+      dst[col*3] = src[col];
+      dst[col*3 + 1] = src[col + realTileWidth];
+      dst[col*3 + 2] = src[col + realTileWidth*2];
+    }
+  } else {
+    union X { unsigned int x; unsigned char c; };
+    if (((union X){1}).c) {
+    for (size_t col = 0; col < tileWidth; ++col) {
+      for (size_t byte = 0; byte < bytesps; ++byte)
+        dst[col*bytesps + byte] = src[col + realTileWidth*(bytesps-byte-1)];  // Little endian
+    }
+    } else {
+    for (size_t col = 0; col < tileWidth; ++col) {
+      for (size_t byte = 0; byte < bytesps; ++byte)
+        dst[col*bytesps + byte] = src[col + realTileWidth*byte];
+        }
+      }
+    }
+}
+
+// From DNG SDK dng_utils.h
```

Comment:
From DNG SDK license agreement:

1. LICENSE GRANT
Software License.  Subject to the restrictions below and other terms of this Agreement, Adobe hereby grants you a non-exclusive, worldwide, royalty free license to use, reproduce, prepare derivative works from, publicly display, publicly perform, distribute and sublicense the Software for any purpose.

2. RESTRICTIONS AND OWNERSHIP
You will not remove any copyright or other notice included in the Software or Documentation and you will include such notices in any copies of the Software that you distribute in human-readable format.

I'll move it to another file and will add Adobe copyright note. Looks like RawTherapee violates Adobe license.